src/backend_tcp/sw/backend_tcp.c

```diff
@@ -293,7 +295,7 @@ int gb_tcp_read_b(struct glip_ctx *ctx, uint32_t channel, size_t size,
         }
         size_read_tmp += sr;
 
-        if (sr != size_remaining) {
+        if ((sr != size_remaining) && (timeout > 0)) {
```

Comment:
That's wrong? If timeout == 0, read_b should retry indefinitely until all requested data has been read. With this change, it tries to read just once and then still returns -ETIMEDOUT. What should be removed instead is the checking of the timeout in line 301 ff